File: src/api/data_api.py
# src/api/data_api.py
from src.facade.CSVReaderFacade import CSVReaderFacade

class DataAPI:
    def __init__(self, filename):
        self.facade = CSVReaderFacade(filename)

    def get_all_data(self):
        return self.facade.get_all_data()

File: src/CPlus/api/DataApi.cpp
#pragma once
#include "../data/CSVReaderFacade.cpp"
#include <iostream>

class DataApi {
private:
    CSVReaderFacade facade;

public:
    DataApi(const std::string& filename) : facade(filename) {
//        std::cout << "Inside DataApi BP1" << filename << std::endl;
}

    std::vector<CSVRow> getAllData() {
//        std::cout << "Inside DataApi BP2" << std::endl;
        return facade.getAllData();
    }

    CSVRow* getByCountryCode(const std::string& countryCode) {
        return facade.getByCountryCode(countryCode);
    }
};
